Cannot find module 'vite-plugin-node-polyfills/shims/buffer'

Just updates to `0.19.0` from `0.16.0` and started getting the error `Cannot find module 'vite-plugin-node-polyfills/shims/buffer`

Downgraded back to `0.16.0` and error vanished. Apologies that I don't have the time atm to put together a reproduction, but it seems to affect every vite v4.x project I have going.

I am getting a similar error when upgrading from `0.17.0` to `0.18.0` and later. I am using `vite@5.0.11`.

```
Error:  Failed to scan for dependencies from entries:
  C:/xxx/frontend/index.html

  X [ERROR] No matching export in "node_modules/vite-plugin-node-polyfills/shims/buffer/dist/index.cjs" for import "Buffer"

    script:C:/xxx/frontend/src/components/renderers/typeHandlers/text/TextEditor.vue?id=0:17:9:
      17 │ import { Buffer } from 'node:buffer'
```

Here is the Stackblitz:  
https://stackblitz.com/edit/stackblitz-starters-8ajb44?file=package.json

```
npm run dev:frontend
```

will yield:

```
Error:   Failed to scan for dependencies from entries:
  /home/projects/stackblitz-starters-8ajb44/frontend/index.html

  ✘ [ERROR] No matching export in "../node_modules/vite-plugin-node-polyfills/shims/buffer/dist/index.cjs" for import "Buffer"

    script:/home/projects/stackblitz-starters-8ajb44/frontend/src/App.vue?id=0:2:9:
      2 │ import { Buffer } from 'node:buffer'
        ╵          ~~~~~~


    at failureErrorWithLog (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:1641:15)
    at eval (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:1049:25)
    at runOnEndCallbacks (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:1476:45)
    at buildResponseToResult (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:1047:7)
    at eval (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:1059:9)
    at requestCallbacks.on-end (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:1058:54)
    at handleRequest (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:722:17)
    at handleIncomingPacket (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:747:7)
    at Socket.readFromStdout (/home/projects/stackblitz-starters-8ajb44/node_modules/esbuild/lib/main.js:670:7)
```

With this change, it will work:

```
npm add -D vite-plugin-node-polyfills@0.17.0 -w frontend
npm run dev:frontend
```

The error can also be removed by converting the package in the `common` workspace from `"type": "commonjs"` to `"type": "module"` (and changing the export syntax in common/src/index.js).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -29,7 +29,7 @@ jobs:
       - run: pnpm build:core
       - run: pnpm build:shims
       - run: pnpm playwright install --with-deps
-      - run: pnpm -r test
+      - run: pnpm -r test:e2e
   test-unit:
     needs:
       - install-dependencies
@@ -39,7 +39,7 @@ jobs:
       - uses: ./.github/actions/install-dependencies
       - run: pnpm build:core
       - run: pnpm build:shims
-      - run: pnpm test:build
+      - run: pnpm -r test
   typecheck:
     needs:
       - install-dependencies

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -1,3 +1,4 @@
 {
-  "typescript.tsdk": "node_modules/typescript/lib"
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "files.eol": "\n"
 }

--- a/examples/vanilla/package.json
+++ b/examples/vanilla/package.json
@@ -4,9 +4,9 @@
   "scripts": {
     "build": "vite build",
     "dev": "vite",
-    "test": "CI=true run-p test:build test:dev",
-    "test:build": "VITE_COMMAND=build WEB_SERVER_COMMAND='vite build && vite preview --port 15174' WEB_SERVER_URL='http://localhost:15174' playwright test",
-    "test:dev": "VITE_COMMAND=dev WEB_SERVER_COMMAND='vite dev --port 15173' WEB_SERVER_URL='http://localhost:15173' playwright test",
+    "test:e2e": "CI=true run-p test:e2e:*",
+    "test:e2e:build": "VITE_COMMAND=build WEB_SERVER_COMMAND='vite build && vite preview --port 15174' WEB_SERVER_URL='http://localhost:15174' playwright test",
+    "test:e2e:dev": "VITE_COMMAND=dev WEB_SERVER_COMMAND='vite dev --port 15173' WEB_SERVER_URL='http://localhost:15173' playwright test",
     "typecheck": "tsc"
   },
   "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -73,7 +73,7 @@
     "build:types": "run-s typecheck",
     "lint": "eslint .",
     "lint:fix": "eslint --fix .",
-    "test": "run-p test:build test:unit",
+    "test": "run-p test:build test:error-repros test:unit",
     "test:build": "run-p test:build:*",
     "test:build:react": "pnpm -C ./examples/react run build",
     "test:build:vanilla": "pnpm -C ./examples/vanilla run build",
@@ -81,6 +81,8 @@
     "test:dev:react": "pnpm -C ./examples/react run dev",
     "test:dev:vanilla": "pnpm -C ./examples/vanilla run dev",
     "test:dev:vue": "pnpm -C ./examples/vue run dev",
+    "test:error-repros": "run-p test:error-repros:*",
+    "test:error-repros:vite-scan-buffer-import-error": "pnpm --filter vite-scan-buffer-import-error test",
     "test:unit": "vitest run --dir ./test",
     "typecheck": "run-p typecheck:*",
     "typecheck:core": "tsc",
@@ -111,7 +113,7 @@
     "vite-plugin-externalize-deps": "^0.1.5",
     "vite-plugin-inspect": "^0.7.42",
     "vite-plugin-node-polyfills": "workspace:*",
-    "vitest": "^1.1.0"
+    "vitest": "^1.2.2"
   },
   "publishConfig": {
     "access": "public"

--- a/pnpm-workspace.yaml
+++ b/pnpm-workspace.yaml
@@ -1,3 +1,4 @@
 packages:
   - .
   - './examples/*'
+  - './test/error-repros/*'

--- a/shims/buffer/package.json
+++ b/shims/buffer/package.json
@@ -8,5 +8,6 @@
     }
   },
   "main": "./dist/index.cjs",
-  "module": "./dist/index.js"
+  "module": "./dist/index.js",
+  "browser": "./dist/index.js"
 }

--- a/shims/global/package.json
+++ b/shims/global/package.json
@@ -8,5 +8,6 @@
     }
   },
   "main": "./dist/index.cjs",
-  "module": "./dist/index.js"
+  "module": "./dist/index.js",
+  "browser": "./dist/index.js"
 }

--- a/shims/process/package.json
+++ b/shims/process/package.json
@@ -8,5 +8,6 @@
     }
   },
   "main": "./dist/index.cjs",
-  "module": "./dist/index.js"
+  "module": "./dist/index.js",
+  "browser": "./dist/index.js"
 }

--- a/test/error-repros/vite-scan-buffer-import-error/index.html
+++ b/test/error-repros/vite-scan-buffer-import-error/index.html
@@ -1,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>vite-scan-buffer-import-error</title>
+  </head>
+  <body>
+    <div id="app"></div>
+    <script type="module" src="/test.ts"></script>
+  </body>
+</html>

--- a/test/error-repros/vite-scan-buffer-import-error/package.json
+++ b/test/error-repros/vite-scan-buffer-import-error/package.json
@@ -1,0 +1,12 @@
+{
+  "name": "vite-scan-buffer-import-error",
+  "type": "module",
+  "private": true,
+  "scripts": {
+    "test": "vite optimize"
+  },
+  "devDependencies": {
+    "vite-plugin-node-polyfills": "workspace:*",
+    "vite@5.1.0": "npm:vite@5.1.0"
+  }
+}

--- a/test/error-repros/vite-scan-buffer-import-error/test.ts
+++ b/test/error-repros/vite-scan-buffer-import-error/test.ts
@@ -1,0 +1,4 @@
+import { Buffer } from 'node:buffer'
+
+// eslint-disable-next-line no-console
+console.log(Buffer.from('hello').toString())

--- a/test/error-repros/vite-scan-buffer-import-error/vite.config.ts
+++ b/test/error-repros/vite-scan-buffer-import-error/vite.config.ts
@@ -1,0 +1,10 @@
+import { nodePolyfills } from 'vite-plugin-node-polyfills'
+
+export default {
+  optimizeDeps: {
+    force: true,
+  },
+  plugins: [
+    nodePolyfills(),
+  ],
+}

